### ISSUE TITLE
Are there any plans to add imperative binding support to out of process .Net 5 functions?

### ISSUE THREAD

**Issue by Jay-Madden** [Azure/azure-functions-dotnet-worker#303]
If this question has already been answered somewhere or is already being tracked/is in the wrong place please let me know.

**Comment by fabiocav** [comment]
@Jay-Madden that's a great question!

Our plan is to ensure there's a model with the right extensibility points to support this, and that is the case, but until we go through a formal API review for all the different extensibility APIs, a lot of that functionality is internal.

Though we have discussed introducing an imperative model, the initial goal is to exercise the extensibility points and identify potential gaps there. We'd likely make it publicly available, but there are no immediate plans to GA and add official support for it unless there's enough demand. This is mainly something we're adding to enable some key validation scenarios and to enable to community to replace the default attributed model if needed.

**Comment by Jay-Madden** [comment]
Noted. Until its added, no imperative binding is a total blocker on our front in any attempt to upgrade many of our FAs.

**Comment by fabiocav** [comment]
That's good feedback. What model are you using today?

**Comment by fabiocav** [comment]
@anthonychu for awareness

**Comment by Jay-Madden** [comment]
we use imperative/runtime blob storage bindings if im understanding your question correctly

**Comment by fabiocav** [comment]
Makes sense. For clarity (for others coming to the issue), we're talking about the `IBinder` model documented here: https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-class-library?tabs=v2%2Ccmd#binding-at-runtime

**Comment by SeanFeldman** [comment]
Being able to change the queue name w/o hard-coding queue name would be helpful. Are there any considerations for imperative binding support being exposed by the worker SDK after a year? Or is there an alternative that is documented, @fabiocav ?

**Comment by brettsam** [comment]
Assigning to @fabiocav for a follow-up answer.

**Comment by fabiocav** [comment]
It has been a while since we’ve shared an update here, so I wanted to provide one. At this time, there are no immediate plans to reintroduce or support imperative bindings. The Azure SDKs have evolved significantly since imperative bindings were first introduced, and our recommendation is to leverage the modern SDK features wherever possible.

That said, we understand that some scenarios might still benefit from imperative bindings. If you run into specific cases that aren’t fully addressed by the current SDK feature set, please let us know. We’ll continue tracking these requests and will either consider bringing imperative bindings to our out-of-process model or, preferably, enhance the SDK to fill any gaps we uncover.

We’re closing this issue, but please feel free to open a new issue or comment on related issues if you have further feedback. Thank you for your patience and help!
